### PR TITLE
Feature/map alert

### PR DESCRIPTION
**Pull request by GabrielMoreschi**

### CODE REVIEW

**Comment by GabrielMoreschi** [comment]
[ALERTA URGENTE - Busca, aquisição e apresentação dos dados em mapa](https://trello.com/c/YLe8WVnE/22-alerta-urgente-busca-aquisi%C3%A7%C3%A3o-e-apresenta%C3%A7%C3%A3o-dos-dados-em-mapa)

File: components/support/SupportLayerItemFire.vue [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            let wmsUrl = ''
+            if (this.layer.layer_type === 'wms') {
+                if (this.layer.filters) {
+                    wmsUrl =
+                        this.layer.wms.geoserver.wms_url +
+                        '&env=percentage:' +
+                        this.layer.opacity / 100
+
+                    const filters = this.layer.filters
+                    const filter_alias = 'dt_foco_calor'
```
Comment (dagnaldo) [inline components/support/SupportLayerItemFire.vue:109]: Why is `filter_alias` is defined as **default** hardcoded and not from API?

File: components/urgent-alerts/AlertLayers.vue [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            </l-popup>
+
+            <l-geo-json
+                v-if="!isVectorGrid && featuresLoaded"
+                :geojson="features"
+                :options="{ onEachFeature }"
+                @ready="onAlertReady"
+            />
+
+            <!-- <BaseMetadataPopup
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:26]: Remove comments.

File: components/urgent-alerts/AlertLayers.vue [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+    data: () => ({
+        selectedAlertFeature: null,
+        heatmapLayer: null,
+
+        isVectorGrid: process.env.MONITORING_VECTOR2TILES === 'true',
+        vectorGrid: null,
+
+        style: {
+            CR: {
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:78]: Attributes as strings.

Ex.: 
`"CR": {...}`.

File: components/urgent-alerts/AlertLayers.vue [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+            switch (no_estagio) {
+                case 'CR':
+                    return this.style.CR
+                case 'DR':
+                    return this.style.DR
+                case 'FF':
+                    return this.style.FF
+                case 'DG':
+                    return this.style.DG
+                default:
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:162]: Missing Default.

File: components/urgent-alerts/AlertLayers.vue [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+                                case 'DG':
+                                    return 2
+                                default:
+                                    return 1
+                            }
+                        },
+                    })
+                    .on('click', (e) => {
+                        this.getFeatureDetails(e.layer.properties.id)
+                        // this.$nextTick(() => {
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:202]: Remove.

File: components/urgent-alerts/AlertLayers.vue [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+                this.map.fitBounds(
+                    this.$refs.alertPolygons.mapObject.getBounds()
+                )
+                this.createMonitoramentoHeatLayer()
+            }
+        },
+
+        async getFeatureDetails(featureId) {
+            this.selectedAlertFeature = null
+            // this.$nextTick(() => {
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:231]: Remove?

File: components/urgent-alerts/AlertLayers.vue [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+            //         // )
+            //         .setContent(this.$refs.popupComponent.$el.innerHTML)
+            // })
+
+            try {
+                this.selectedAlertFeature = await this.$api.$get(
+                    'alerts/detail/' + featureId + '/'
+                )
+
+                // this.$nextTick(() => {
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:244]: Remove?

File: components/urgent-alerts/AlertLayers.vue [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+                //         // .bindPopup(
+                //         //     this.$refs.popupComponent.$el
+                //         // )
+                //         .setContent(this.$refs.popupComponent.$el.innerHTML)
+                // })
+            } catch (exception) {
+                this.$store.commit('alert/addAlert', {
+                    message: this.$t('detail-api-error'),
+                })
+                // return null
```
Comment (dagnaldo) [inline components/urgent-alerts/AlertLayers.vue:258]: Remove?

Comment (GabrielMoreschi) [reply components/urgent-alerts/AlertLayers.vue:258]: Resolved

File: components/urgent-alerts/AlertLayers.vue (hunk 6 above)
Comment (GabrielMoreschi) [reply components/urgent-alerts/AlertLayers.vue:231]: Resolved

File: components/urgent-alerts/AlertLayers.vue (hunk 5 above)
Comment (GabrielMoreschi) [reply components/urgent-alerts/AlertLayers.vue:202]: Resolved

File: components/urgent-alerts/AlertLayers.vue (hunk 3 above)
Comment (GabrielMoreschi) [reply components/urgent-alerts/AlertLayers.vue:78]: Resolved

File: components/urgent-alerts/AlertLayers.vue (hunk 2 above)
Comment (GabrielMoreschi) [reply components/urgent-alerts/AlertLayers.vue:26]: Resolved

File: components/land-use/LandUseLayers.vue [hunk 9]
```diff
@@ -75,10 +75,36 @@ export default {
         vectorGrid: null,
 
         style: {
-            weight: 2.5,
-            color: '#8e391b',
-            fill: true,
-            fillOpacity: 1,
+            VI: {
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/land-use/LandUseLayers.vue:78]: Resolved

File: components/land-use/LandUseLayers.vue [hunk 10]
```diff
@@ -117,6 +159,22 @@ export default {
...
+                    return this.style.VI
+                case 'CR':
+                    return this.style.CR
+                case 'MA':
+                    return this.style.MA
+                case 'AG':
+                    return this.style.AG
+                case 'VN':
+                    return this.style.VN
+                default:
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/land-use/LandUseLayers.vue:174]: Resolved

**Review by dagnaldo — APPROVED** [review]

**Pull request merged**